src/App.tsx

```diff
@@ -14,11 +14,94 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  items: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    items: ['Jam'],
+  };
+
+  selected = (good: string) => {
+    this.setState((prevState) => ({
+      items: [...prevState.items, good],
+    }));
+  };
+
+  unselected = () => {
+    this.setState({ items: [] });
+  };
+
+  removeSelect = (selectedItem: string) => {
+    this.setState((prevState) => ({
+      items: prevState.items.filter(good => good !== selectedItem),
+    }));
+  };
+
+  renderBoxGoods = () => {
+    const { items } = this.state;
+
+    if (items.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (items.length === 1) {
+      return `${items[0]} is selected`;
+    }
+
+    return `${items.slice(0, -1).join(', ')}
+          and ${items.slice(-1)} are selected`;
+  };
+
+  render(): React.ReactNode {
+    const { items } = this.state;
+
+    return (
+      <div className="container is-max-desktop">
+
+        <h1 className="title">
+          {this.renderBoxGoods()}
+        </h1>
+
+        <ul className="display">
+          {goodsFromServer.map((good) => {
+            return (
+              <li
+                key={good}
+                className={`list ${items.includes(good)
+                  ? 'list__active' : ''}`}
+              >
+                {good}
+
+                <button
+                  type="button"
+                  className="button"
+                  onClick={
+                    items.includes(good)
```

Comment:
You did this check at least 3 times in your code
You can create one separate variable before return statement in map and save the result of this method here